### PR TITLE
IGNITE-10184: Fixed type conflict check test in public schema.

### PR DESCRIPTION
**Pull request by aliskhakov**

### CODE REVIEW

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSchemaSelfTest.java [hunk 1]
```diff
@@ -292,19 +294,19 @@ private void testQueryEntity(IgniteCache<Long, Person> cache, String tbl) {
...
-            .setName(CACHE_PERSON)
-            .setIndexedTypes(PersonKey.class, Person.class)
-            .setSqlSchema(QueryUtils.DFLT_SCHEMA));
-
-        node.createCache(new CacheConfiguration<PersonKey, Person>()
-            .setName(CACHE_PERSON_2)
-            .setIndexedTypes(PersonKey.class, Person.class)
-            .setSqlSchema(QueryUtils.DFLT_SCHEMA));
+    public void testTypeConflictInPublicSchema() throws Exception {
+	    node.createCache(new CacheConfiguration<PersonKey, Person>()
```
Comment (SomeFire) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSchemaSelfTest.java]: Use 4 spaces instead of tabs.

Comment (aliskhakov) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSchemaSelfTest.java]: Done.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSchemaSelfTest.java [hunk 2]
```diff
@@ -292,19 +294,19 @@ private void testQueryEntity(IgniteCache<Long, Person> cache, String tbl) {
...
-            .setName(CACHE_PERSON_2)
-            .setIndexedTypes(PersonKey.class, Person.class)
-            .setSqlSchema(QueryUtils.DFLT_SCHEMA));
+        GridTestUtils.assertThrows(log, (Callable<Void>) () -> {
+            node.createCache(new CacheConfiguration<PersonKey, Person>()
+                .setName(CACHE_PERSON_2)
+                .setIndexedTypes(PersonKey.class, Person.class)
+                .setSqlSchema(QueryUtils.DFLT_SCHEMA));
+            return null;
+        }, CacheException.class, "Table already exists: PERSON");
```
Comment (alex-plekhanov) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSchemaSelfTest.java]: Perhaps it's better to check not error text, but error code. Error text can be changed in future, error code most likely remain the same. You can extract underlying `SchemaOperationException` (use `X.cause()` method) and check `code` field to equal `SchemaOperationException.CODE_TABLE_EXISTS`

Comment (aliskhakov) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSchemaSelfTest.java]: Done.

**Pull request closed without merging**